src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -246,6 +247,16 @@ function reducer(draft, action) {
       );
       break;
     }
+    case 'REUSABLE_FORM_DEFINITIONS_LOADED': {
+      const reusableFormDefinitions = action.payload;
+      const formDefinitionsIds = draft.formDefinitions.map(fd => fd.uuid);
+      draft.formDefinitions = [
+        ...draft.formDefinitions,
+        ...reusableFormDefinitions.filter(fd => !formDefinitionsIds.includes(fd.uuid)),
+      ];
```

Comment:
I am wondering if there's a possible state update race condition? if both requests (used in form, reusable FDs) complete at the same time because they have parallel requests, we have two actions writing to the same state key, maybe one overwrites the other?

Reply:
That might be the reason

Reply:
I'd like to take some time today to dive into this myself and try to figure out what's going on

Reply:
Well turns out it might be a race condition issue:

`test_all_reuseable_form_definition_options_are_available_when_not_part_of_the_form`:
https://github.com/open-formulieren/open-forms/blob/632c23d78bec16f16ce4542b0bc276f6ae4b0920/src/openforms/js/components/admin/form_design/NewStepFormDefinitionPicker.js#L23
https://github.com/open-formulieren/open-forms/blob/632c23d78bec16f16ce4542b0bc276f6ae4b0920/src/openforms/forms/tests/e2e_tests/test_form_designer.py#L1528


test output:

`[]`

Locally:

`[Object, Object]`

Reply:
Fixed by last two commits